### PR TITLE
[user-authn] Add demo files

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Why do we need it, and what problem does it solve?
Add an instruction on how to test the user-authn module customization features

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: user-authn
type: chore
summary: Add files for user-authn demo
impact_level: low
```

### CODE REVIEW

File: modules/150-user-authn/docs/internal/demo/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Steps
+
+## Install OpenLDAP
+
+* Apply [**ldap**](./ldap.yaml) manifests to the cluster.
```
Comment (z9r5) [inline modules/150-user-authn/docs/internal/demo/README.md:5]:
```suggestion
* Apply [**ldap**](./ldap.yaml) manifest to the cluster.
```
?

File: modules/150-user-authn/docs/internal/demo/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Add DexProvider
+
+* Apply [**dex-provider**](./dex-provider.yaml) manifest to the cluster.
+* Login to Kubeconfig (username: `janedoe@example.com`, password: `foo` or username: `johndoe@example.com`, password: `bar`)
+
+## Generate kubeconfig
+
+* Copy kubeconfig to your PC
+* Show how kubeconfig works
```
Comment (z9r5) [inline modules/150-user-authn/docs/internal/demo/README.md:15]:
```suggestion
* Login to Kubeconfig (username: `janedoe@example.com`, password: `foo` or username: `johndoe@example.com`, password: `bar`).

## Generate kubeconfig

* Copy kubeconfig to your PC.
* Show how kubeconfig works.
```

File: modules/150-user-authn/docs/internal/demo/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+* Show that access is protected (`janedoe@example.com` can access echo server, `johndoe@example.com` cannot).
+
+## Create a user
+
+* Apply [**dex-user**](./dex-user.yaml) manifest to the cluster.
+* Show that you can log in with credentials from the custom resource.
+* Add `groups: ["developers"]` to the User spec to show that this user now has access to the echo server. 
+```shell
+kubectl patch user openldap-demo --type='merge' -- patch '{"spec": {"groups": ["developers"]}}'
+```
```
Comment (z9r5) [inline modules/150-user-authn/docs/internal/demo/README.md:41]: Please add 2 indent to the code block.

File: modules/150-user-authn/docs/internal/demo/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+* Show that access is protected (`janedoe@example.com` can access echo server, `johndoe@example.com` cannot).
+
+## Create a user
+
+* Apply [**dex-user**](./dex-user.yaml) manifest to the cluster.
+* Show that you can log in with credentials from the custom resource.
+* Add `groups: ["developers"]` to the User spec to show that this user now has access to the echo server. 
+```shell
+kubectl patch user openldap-demo --type='merge' -- patch '{"spec": {"groups": ["developers"]}}'
```
Comment (nabokihms) [inline modules/150-user-authn/docs/internal/demo/README.md:40]:
```suggestion
  kubectl patch user openldap-demo --type='merge' -- patch '{"spec": {"groups": ["developers"]}}'
```

File: modules/150-user-authn/docs/internal/demo/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+kubectl -n openldap-demo annotate ingress echoserver 'nginx.ingress.kubernetes.io/auth-url=https://echoserver-dex-authenticator.openldap-demo.svc.cluster.local/dex-authenticator/auth'
+```
+* Show that access is protected (`janedoe@example.com` can access echo server, `johndoe@example.com` cannot).
+
+## Create a user
+
+* Apply [**dex-user**](./dex-user.yaml) manifest to the cluster.
+* Show that you can log in with credentials from the custom resource.
+* Add `groups: ["developers"]` to the User spec to show that this user now has access to the echo server. 
+```shell
```
Comment (nabokihms) [inline modules/150-user-authn/docs/internal/demo/README.md:39]:
```suggestion
  ```shell
```

File: modules/150-user-authn/docs/internal/demo/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+kubectl patch user openldap-demo --type='merge' -- patch '{"spec": {"groups": ["developers"]}}'
+```
```
Comment (nabokihms) [inline modules/150-user-authn/docs/internal/demo/README.md:41]:
```suggestion
  ```
```

**Review by z9r5 — APPROVED** [review]

**Pull request merged**